security-baseline/rule-types/github/osps-do-01.yaml

```
@@ -0,0 +1,54 @@
+version: v1
+release_phase: alpha
+type: rule-type
+name: osps-do-01
+display_name: The project has mechanisms for public discussion
+short_failure_message: The project does not publicize mechanisms for public discussion.
+severity:
+  value: info
+context:
+  provider: github
+description: |
+  Projects should encourage open communication and collaboration
+  within the their community, enabling users to provide feedback and
+  discuss proposed changes or usage challenges.
+guidance: |
+  Enable "Issues" or "Discussions" via the
+  [Settings](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository)
+  page on GitHub.
+def:
+  in_entity: repository
+  rule_schema: {}
+  ingest:
+    type: git
+  eval:
+    type: rego
+    data_sources:
+      - name: ghapi
+    rego:
+      type: deny-by-default
+      def: |
+        package minder
+
+        import future.keywords.every
+        import future.keywords.if
+
+        default allow := false
+
+        allow if {
+          repo := minder.datasource.ghapi.repo({
+            "owner": input.properties["github/repo_owner"],
+            "repo": input.properties["github/repo_name"]
+          })
```

I'm assuming that Rego can cache these identical datasource queries?  If not, it might make sense to move them outside the allow:

```suggestion
        repo := minder.datasource.ghapi.repo({
          "owner": input.properties["github/repo_owner"],
          "repo": input.properties["github/repo_name"]
        })
        allow if { repo.body.has_issues }
        allow if { repo.body.has_discussions }
```

---

You're right, it is better to move that out, I tried that but it was failing for another reason (I was using `repo.has_issues` rather than`repo.body.has_issues`).

I'll fix this.